extensions/kubeapi/secrets/update.go

```diff
@@ -0,0 +1,40 @@
+package secrets
+
+import (
+	"context"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/unstructured"
+	"github.com/rancher/shepherd/pkg/api/scheme"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// UpdateSecret is a helper function that uses the dynamic client to update a secret in a specific cluster.
+func UpdateSecret(client *rancher.Client, clusterID string, existingSecret *corev1.Secret, updatedSecret *corev1.Secret) (*corev1.Secret, error) {
+	dynamicClient, err := client.GetDownStreamClusterClient(clusterID)
+	if err != nil {
+		return nil, err
+	}
+
+	secretResource := dynamicClient.Resource(SecretGroupVersionResource).Namespace(existingSecret.Namespace)
+	secretUnstructured, err := secretResource.Get(context.TODO(), existingSecret.Name, metav1.GetOptions{})
```

Comment:
I've got the same question here

Reply:
I made recent changes to this function to accept an existing object, and did not realize that I should remove the GET request here.